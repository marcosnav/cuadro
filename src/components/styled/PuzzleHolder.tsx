import styled from 'styled-components';

const PuzzleHolder = styled.div`
  background: ${(props) => props.theme.GD_BLUE};
  background: ${({ theme }) => `linear-gradient(30deg, ${theme.GD_BLUE} 0%, ${theme.GD_PURPLE} 100%)`};
  box-shadow: inset 0 1px 4px 2px rgba(0, 0, 0, 0.1);
  height: 268px;
  width: 268px;
`;

PuzzleHolder.displayName = 'PuzzleHolder';

export default PuzzleHolder;
