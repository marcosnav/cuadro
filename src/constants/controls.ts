/**
 * Available control types
 */
export enum ControlType {
  NEW_GAME = 'NEW_GAME',
  RESTART_GAME = 'RESTART_GAME',
  SEE_ORIGINAL = 'SEE_ORIGINAL',
}
