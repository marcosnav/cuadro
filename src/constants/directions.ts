/**
 * SwipeDirection, one of UP, RIGHT, LEFT, DOWN
 */
export enum SwipeDirection {
  UP = 'UP',
  RIGHT = 'RIGHT',
  DOWN = 'DOWN',
  LEFT = 'LEFT',
}
