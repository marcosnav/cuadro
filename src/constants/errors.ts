export enum errors {
  NO_INTERNET = 'NO_INTERNET',
  UNEXPECTED_ERROR = 'UNEXPECTED_ERROR',
  UNRESPONSIVE_IMAGE_PROVIDER = 'UNRESPONSIVE_IMAGE_PROVIDER',
}
