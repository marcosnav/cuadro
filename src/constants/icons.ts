import check from './../assets/svg/check.svg';
import image from './../assets/svg/image.svg';
import newicon from './../assets/svg/new.svg';
import refresh from './../assets/svg/refresh.svg';

export const icons: { [key: string]: string } = {
  check,
  image,
  new: newicon,
  refresh,
};
