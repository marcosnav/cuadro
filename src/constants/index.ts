export * from './status';
export * from './errors';
export * from './theme';
export * from './keycodes';
