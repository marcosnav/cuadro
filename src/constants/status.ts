/**
 * App Status Scenarios
 */
export enum Status {
  STARTING_NEW_GAME = 'STARTING_NEW_GAME',
  LOADING_IMAGE = 'LOADING_IMAGE',
  DISPLAY_ORIGINAL = 'DISPLAY_ORIGINAL',
  PLAYING_GAME  = 'PLAYING_GAME',
  FINISHED_GAME = 'FINISHED_GAME',
  ERROR = 'ERROR',
}
