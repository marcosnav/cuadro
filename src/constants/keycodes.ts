import { SwipeDirection } from './directions';

/**
 * KeyCodes to match for keypress events
 */
export enum KeyCodes {
  DOWN = 40,
  LEFT = 37,
  RIGHT = 39,
  UP = 38,
  ENTER = 13,
}

/**
 * Map key code with swipe direction
 */
export const KeySwipeMap: { [key: number]: SwipeDirection } = {
  40: SwipeDirection.DOWN,
  37: SwipeDirection.LEFT,
  39: SwipeDirection.RIGHT,
  38: SwipeDirection.UP,
};
