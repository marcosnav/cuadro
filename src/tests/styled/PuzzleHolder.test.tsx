import { mount } from 'enzyme';
import 'jest-styled-components';
import React from 'react';
import { ThemeConsumer } from 'styled-components';
import PuzzleHolder from './../../components/styled/PuzzleHolder';
import { theme } from './../../constants';

describe('<PuzzleHolder/>', () => {
  beforeAll(() => {
    ThemeConsumer._currentValue = theme;
  });

  test('renders properly', () => {
    const component = mount(<PuzzleHolder />);
    expect(component).toHaveStyleRule('background', theme.SOFT_GRAY);
    expect(component).toHaveStyleRule('height', '268px');
    expect(component).toHaveStyleRule('width', '268px');
    expect(component).toMatchSnapshot();
  });
});
