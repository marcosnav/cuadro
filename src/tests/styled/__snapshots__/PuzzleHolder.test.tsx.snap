// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PuzzleHolder/> renders properly 1`] = `
.c0 {
  background: #238BFF;
  background: linear-gradient(30deg,#238BFF 0%,#7D05AA 100%);
  box-shadow: inset 0 1px 4px 2px rgba(0,0,0,0.1);
  height: 268px;
  width: 268px;
}

<PuzzleHolder>
  <div
    className="c0"
  />
</PuzzleHolder>
`;
